vars/goDefaultVersion.groovy

```diff
@@ -36,3 +36,7 @@ def call(Map args = [:]) {
 def isGoVersionEnvVarSet(){
   return env.GO_VERSION != null && "" != "${env.GO_VERSION}"
 }
+
+def defaultVersion(){
+  return '1.15.6'
```

Comment:
```suggestion
  return staticGoDefaultVersion()
```

Reply:
I do not get this.

Reply:
`step.method` cannot be mocked then it requires to use in the UTs the below snippet:

```
   def script = loadScript('vars/step.groovy')
   def value = script.method()
```

As far as I see `step.method` in our case is not used but only for given dynamism when running the UTs.

If that's the case, I was suggesting:

* create a new step called `staticGoDefaultVersion`
* then mock the step`helper.registerAllowedMethod('staticGoDefaultVersion', ...` in WithGoEnvUnixStepTests

Reply:
finally was simpler https://github.com/elastic/apm-pipeline-library/pull/903#discussion_r557517613